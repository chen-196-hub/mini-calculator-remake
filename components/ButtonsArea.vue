<template>
  <v-card class="text-center">
    <v-row v-for="r in keyContent" :key="r.key" no-gutters>
      <v-col v-for="n in r" :key="n.key" :cols="n.key === '0' ? 6 : 3">
        <v-card class="py-2 px-5" tile>
          <v-btn x-large rounded width="100%" @click="$emit('onClick', n.key)">
            {{ n.value }}
          </v-btn>
        </v-card>
      </v-col>
    </v-row>
  </v-card>
</template>

<script setup lang="ts">
import { ref } from 'vue'
interface KeyContent {
  key: string
  value: string
}

const keyContent = ref<KeyContent[][]>([
  [
    { key: 'ac', value: 'AC' },
    { key: 'plu-min', value: '+/-' },
    { key: 'percent', value: '%' },
    { key: 'divide', value: '‚ûó' },
  ],
  [
    { key: '7', value: '7' },
    { key: '8', value: '8' },
    { key: '9', value: '9' },
    { key: 'times', value: '‚úñÔ∏è' },
  ],
  [
    { key: '4', value: '4' },
    { key: '5', value: '5' },
    { key: '6', value: '6' },
    { key: 'minus', value: '‚ûñ' },
  ],
  [
    { key: '1', value: '1' },
    { key: '2', value: '2' },
    { key: '3', value: '3' },
    { key: 'plus', value: '‚ûï' },
  ],
  [
    { key: '0', value: '0' },
    { key: 'ten', value: '.' },
    { key: 'equal', value: 'üü∞' },
  ],
])
</script>
