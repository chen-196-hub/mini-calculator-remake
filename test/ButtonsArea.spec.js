import Vuetify from 'vuetify'
// Utilities
import { createLocalVue, mount } from '@vue/test-utils'
import ButtonsArea from '@/components/ButtonsArea.vue'

const localVue = createLocalVue()

describe('ButtonsArea.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const mountFunction = (options) => {
    return mount(ButtonsArea, {
      localVue,
      vuetify,
      ...options,
    })
  }

  it('should match snapshot', () => {
    const wrapper = mountFunction()

    expect(wrapper.html()).toMatchSnapshot()
  })

  // it('should been called emit', () => {
    
  // })

})
