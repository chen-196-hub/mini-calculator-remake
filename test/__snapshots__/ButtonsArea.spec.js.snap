// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsArea.vue should match snapshot 1`] = `"<div class="text-center v-card v-sheet theme--light"></div>"`;
